components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/APIArtifactGeneratorUtil.java

```diff
@@ -51,6 +51,9 @@ public static RuntimeArtifactDto generateAPIArtifact(List<String> apiUuids, Stri
                     gatewayArtifacts =
                             gatewayArtifactsMgtDAO.retrieveGatewayArtifactsByLabel(gatewayLabels, organization);
                 }
+            } else if (!apiUuids.isEmpty()) {
```

Comment:
Check null before checking empty. (No need if null check happen in any earlier phase)

Reply:
Fixed: https://github.com/wso2/carbon-apimgt/pull/11304/commits/c1dd6055755e9d451636e15a814c7aff415a99e1